app/controllers/replies_controller.rb

```diff
@@ -110,6 +110,23 @@ def create
       @unseen_replies = reply.post.replies.ordered.paginate(page: 1, per_page: 10)
       @unseen_replies = @unseen_replies.where('reply_order > ?', last_seen_reply_order) if last_seen_reply_order.present?
       most_recent_unseen_reply = @unseen_replies.last
+
+      if params[:allow_dupe].blank?
+        last_by_user = reply.post.replies.where(user_id: reply.user_id).ordered.last
+        match_attrs = ['content', 'icon_id', 'character_id', 'character_alias_id']
+        if last_by_user.present? && last_by_user.attributes.slice(*match_attrs) == reply.attributes.slice(*match_attrs)
+          flash.now[:error] = "This looks like a duplicate. Did you attempt to post this twice? Please resubmit if this was intentional."
+          @allow_dupe = true
+          if @unseen_replies.count == 0 || (@unseen_replies.count == 1 && most_recent_unseen_reply.id == last_by_user.id)
```

Comment:
Why does this use `.count`? On a relation that'll trigger a new query, which seems less ideal than just using the most_recent_unseen_reply object we've already got?

Reply:
Using the object would work for 'does this exist' I guess but doesn't work for 'are there more than one'. I'm not sure if there's a more efficient way to do that?

Reply:
Well, in the case where we expect that @unseen_replies will be loaded into memory anyway, we can use .size or whatever it is that gets "number of memory objects" instead of "SQL query", and in the case where we do not expect this we could at least change the first check? Maybe?

Reply:
Oh hm yeah that's valid @unseen_replies would be loaded into memory if it exists for rendering the page a little after this. Although it's going through replies/_single I think so ti might not be loaded per-say? Anyway we can fix the first check either way.